core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -158,6 +158,37 @@ private[spark] class HostLocalDirManager(
   }
 }
 
+private[spark] trait MigratableRDDResolver {
+
+  def getMigratableRDDBlocks(): Seq[ReplicateBlock]
+
+  def migrateBlock(block: ReplicateBlock): Boolean
+}
+
+private[spark] class BlockManagerPeerProvider(
+    blockManagerId: BlockManagerId,
+    cachedPeersTtlMs: Int,
+    master: BlockManagerMaster) extends Logging {
+
+  // Field related to peer block managers that are necessary for block replication
+  @volatile private var cachedPeers: Seq[BlockManagerId] = _
+  private var lastPeerFetchTimeNs = 0L
+
+  /**
+   * Get peer block managers in the system.
+   */
+  def getPeers(forceFetch: Boolean): Seq[BlockManagerId] = synchronized {
+    val diff = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - lastPeerFetchTimeNs)
```

Comment:
Why did you get rid of the lock?

Reply:
The lock is there. See the `synchronized {` block right in the `def getPeers...` line. In this case the the lock object is the `this` which is as sufficient as the artificial `object` ( the `peerFetchLock`) was: it is only used by this method so the scopes are equal.

Reply:
ah gotcha, now that it's its own class it doesn't need a distinct lock object, k.